File: src/test/java/com/ject/studytrip/trip/fixture/CreateDailyGoalRequestFixture.java
package com.ject.studytrip.trip.fixture;

import com.ject.studytrip.pomodoro.presentation.dto.request.CreatePomodoroRequest;
import com.ject.studytrip.trip.presentation.dto.request.CreateDailyGoalRequest;
import java.util.List;

public class CreateDailyGoalRequestFixture {

    private CreatePomodoroRequest pomodoro = new CreatePomodoroRequest(30);
    private List<Long> missionIds = List.of(1L, 2L);

    public CreateDailyGoalRequestFixture withPomodoro(CreatePomodoroRequest pomodoro) {
        this.pomodoro = pomodoro;
        return this;
    }

    public CreateDailyGoalRequestFixture withMissionIds(List<Long> missionIds) {
        this.missionIds = missionIds;
        return this;
    }

    public CreateDailyGoalRequest build() {
        return new CreateDailyGoalRequest(pomodoro, missionIds);
    }
}

File: src/test/kotlin/com/ject/studytrip/trip/fixture/CreateDailyGoalRequestFixture.kt
package com.ject.studytrip.trip.fixture

import com.ject.studytrip.pomodoro.presentation.dto.request.CreatePomodoroRequest
import com.ject.studytrip.trip.presentation.dto.request.CreateDailyGoalRequest

class CreateDailyGoalRequestFixture(
    private val pomodoro: CreatePomodoroRequest = CreatePomodoroRequest(30, 1),
    private val missionIds: List<Long> = emptyList(),
) {
    fun withPomodoro(pomodoro: CreatePomodoroRequest): CreateDailyGoalRequestFixture = CreateDailyGoalRequestFixture(pomodoro, missionIds)

    fun withMissionIds(missionIds: List<Long>): CreateDailyGoalRequestFixture = CreateDailyGoalRequestFixture(pomodoro, missionIds)

    fun build(): CreateDailyGoalRequest = CreateDailyGoalRequest(pomodoro, missionIds)
}
